From 7b47bfef962fe02f21eefb1defd7149f22b9f820 Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Wed, 4 Jul 2018 16:53:44 +0300
Subject: [PATCH 0656/1051] fix: mmc: xenon: Add delay upon MMC voltage switch

Add function for delaying excution by 5ms after switching
the MMC voltage.
This delay is required by Xenon controller, otherwise voltage
switch 3.3V -> 1.8V always fails, so the interface falls back
to the legacy speed mode.

Change-Id: I1f6c4ef96d1bd8c7ec64182b01164cd74b14c46e
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/mmc/host/sdhci-xenon.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/mmc/host/sdhci-xenon.c b/drivers/mmc/host/sdhci-xenon.c
index f8cf2b4b39a2..36a190dc1709 100644
--- a/drivers/mmc/host/sdhci-xenon.c
+++ b/drivers/mmc/host/sdhci-xenon.c
@@ -249,6 +249,7 @@ static const struct sdhci_ops sdhci_xenon_ops = {
 	.reset			= xenon_reset,
 	.set_uhs_signaling	= xenon_set_uhs_signaling,
 	.get_max_clock		= sdhci_pltfm_clk_get_max_clock,
+	.voltage_switch		= xenon_voltage_switch,
 };
 
 static const struct sdhci_pltfm_data sdhci_xenon_pdata = {
-- 
2.17.1

